### DISCUSSION TITLE
Realtime using service role only? Current best way to do private realtime?

### DISCUSSION THREAD

**Discussion by lsei** [supabase/supabase#3768] [Questions]
Hello! Just getting into supabase and really loving the approach. 

I'm aware there's still work to be done to properly support RLS for the realtime database access. 

Is there any way to privately expose/connect to the current non-rls realtime, for example somehow using the service role token, without enabling it publically, where full RLS support would be required? 

Are there any recommended ways of handling private real time data updates (for example for a chat)?

Thanks for any advice.

**Comment by GaryAustin1** [comment]
I have numerous low volume chat rooms in my app.  I have a status/flag table with an entry per room the gets updated by a trigger when a message is inserted into the RLS protected message table.  I use time, but one could use incrementing counter.  This is totally  non RLS and I do realtime change on the room row.  When I get the change then I go read the message table with RLS security based on room  id and user being signed up for that chat room.

**Reply by lsei** [reply]
Thanks Gary. That seems like an interesting workaround!
